framework/audio/engine/internal/audioengine.cpp

```diff
@@ -167,12 +228,76 @@ void AudioEngine::setPlaySpeed(handle h, float speed)
     m_sl->engine.setRelativePlaySpeed(h, speed);
 }
 
-mu::async::Notification AudioEngine::playCallbackCalled() const
+void AudioEngine::swapPlayContext(handle h, Context& ctx)
 {
-    return m_playCallbackCalled;
+    HandleMeta& m = meta(h);
+    IF_ASSERT_FAILED(m.isValid()) {
+        return;
+    }
+
+    m.playContext.swap(ctx);
+    if (m.playContextRequested) {
+        m.playContextChanged.send(m.playContext);
+    }
+
+    if (m.playContext.hasVal(CtxKey::InstanceDestroyed)) {
+        onStop(h);
+    }
 }
 
-void AudioEngine::onPlayCallbackCalled()
+mu::async::Channel<Context> AudioEngine::playContextChanged(handle h) const
 {
-    m_playCallbackCalled.notify();
+    HandleMeta& m = const_cast<AudioEngine*>(this)->meta(h);
+    IF_ASSERT_FAILED(m.isValid()) {
+        return mu::async::Channel<Context>();
+    }
+    m.playContextRequested = true;
+    return m.playContextChanged;
+}
+
+AudioEngine::HandleMeta& AudioEngine::pushMeta(handle h)
+{
+    std::lock_guard<std::mutex> lock(m_metasMutex);
+    auto it = m_handleMetas.find(h);
+    IF_ASSERT_FAILED(it == m_handleMetas.end()) {
+        return it->second;
+    }
+
+    HandleMeta& m = m_handleMetas[h]; //! NOTE Will be created
```

Comment:
Do you mean that it'll be created as a default value by container?

Reply:
Yes, it will created (default) and added to the container.